algorithmic_efficiency/workloads/wmt/wmt_jax/workload.py

```diff
@@ -192,18 +186,10 @@ def translate_and_calculate_bleu(self,
                                    num_batches,
                                    max_predict_length: int):
     """Translates the `predict_ds` and calculates the BLEU score."""
-    n_devices = jax.local_device_count()
     logging.info('Translating evaluation dataset.')
     sources, references, predictions = [], [], []
     for _ in range(num_batches):
       pred_batch = next(ds_iter)
-      # Handle final odd-sized batch by padding instead of dropping it.
-      cur_pred_batch_size = pred_batch['inputs'].shape[0]
-      if cur_pred_batch_size % n_devices:
-        padded_size = int(np.ceil(cur_pred_batch_size / n_devices) * n_devices)
-        pred_batch = jax.tree_map(
-            functools.partial(_pad_examples, padded_size),  # pylint: disable=cell-var-from-loop
-            pred_batch)
```

Comment:
why do we need to delete this? we have similar logic for pytorch, right?

Reply:
Because the batch is already [sharded](https://github.com/runame/algorithmic-efficiency/blob/pytorch-speedups/algorithmic_efficiency/workloads/wmt/input_pipeline.py#L289) (before by calling `ds = ds.batch(...)`), hence there is no need for padding anymore. For PyTorch without DDP, the batch is not sharded at all, but I just realized that in fact it is first sharded and then [reshaped](https://github.com/runame/algorithmic-efficiency/blob/pytorch-speedups/algorithmic_efficiency/workloads/wmt/wmt_pytorch/workload.py#L259-L260) again, so we can also delete the code block for PyTorch.

Note, that the current way of sharding the batches [assumes](https://github.com/runame/algorithmic-efficiency/blob/pytorch-speedups/algorithmic_efficiency/data_utils.py#L22-L25) that the remainder batch, which is not dropped for evaluation, is evenly divisble by the number of GPUs. This affects all workloads which use this sharding function, e.g. ImageNet and WMT. This might obviously lead to problems, depending on the batch size used in the submissions. One easy solution would be to hard-code the evaluation batch size or provide an error which suggests the closest batch size which works well to the user. For the part where the batch size is more important, i.e. training, this issue does not appear anyway, because the remainder batch is dropped.

Reply:
Maybe a good solution is to provide the option to choose the training and eval batch size separately and provide a working default for evaluation. If the user really wants to change the evaluation batch size, we can raise an error which suggests the closest working one. This also adds more flexibility for the user, since they might want to use a larger batch size for evaluation than for training.

Reply:
Yeah having different batch sizes for train and eval is super useful in general, we should definitely support it. I'll add a TODO for myself to look into it (unless you want to soon!)